influxql/query_executor.go

```diff
@@ -0,0 +1,423 @@
+type QueryExecutor struct {
+	AlterRetentionPolicyStatementExecutor  AlterRetentionPolicyStatementExecutor
+	CreateContinuousQueryStatementExecutor CreateContinuousQueryStatementExecutor
+	CreateDatabaseStatementExecutor        CreateDatabaseStatementExecutor
+	CreateRetentionPolicyStatementExecutor CreateRetentionPolicyStatementExecutor
+	CreateSubscriptionStatementExecutor    CreateSubscriptionStatementExecutor
+	CreateUserStatementExecutor            CreateUserStatementExecutor
+	DeleteStatementExecutor                DeleteStatementExecutor
+	DropContinuousQueryStatementExecutor   DropContinuousQueryStatementExecutor
+	DropDatabaseStatementExecutor          DropDatabaseStatementExecutor
+	DropMeasurementStatementExecutor       DropMeasurementStatementExecutor
+	DropRetentionPolicyStatementExecutor   DropRetentionPolicyStatementExecutor
+	DropSeriesStatementExecutor            DropSeriesStatementExecutor
+	DropServerStatementExecutor            DropServerStatementExecutor
+	DropSubscriptionStatementExecutor      DropSubscriptionStatementExecutor
+	DropUserStatementExecutor              DropUserStatementExecutor
+	GrantAdminStatementExecutor            GrantAdminStatementExecutor
+	GrantStatementExecutor                 GrantStatementExecutor
+	RevokeAdminStatementExecutor           RevokeAdminStatementExecutor
+	RevokeStatementExecutor                RevokeStatementExecutor
+	SelectStatementExecutor                SelectStatementExecutor
+	SetPasswordUserStatementExecutor       SetPasswordUserStatementExecutor
+	ShowContinuousQueriesStatementExecutor ShowContinuousQueriesStatementExecutor
+	ShowDatabasesStatementExecutor         ShowDatabasesStatementExecutor
+	ShowDiagnosticsStatementExecutor       ShowDiagnosticsStatementExecutor
+	ShowFieldKeysStatementExecutor         ShowFieldKeysStatementExecutor
+	ShowGrantsForUserStatementExecutor     ShowGrantsForUserStatementExecutor
+	ShowMeasurementsStatementExecutor      ShowMeasurementsStatementExecutor
+	ShowRetentionPoliciesStatementExecutor ShowRetentionPoliciesStatementExecutor
+	ShowSeriesStatementExecutor            ShowSeriesStatementExecutor
+	ShowServersStatementExecutor           ShowServersStatementExecutor
+	ShowShardGroupsStatementExecutor       ShowShardGroupsStatementExecutor
+	ShowShardsStatementExecutor            ShowShardsStatementExecutor
+	ShowStatsStatementExecutor             ShowStatsStatementExecutor
+	ShowSubscriptionsStatementExecutor     ShowSubscriptionsStatementExecutor
+	ShowTagKeysStatementExecutor           ShowTagKeysStatementExecutor
+	ShowTagValuesStatementExecutor         ShowTagValuesStatementExecutor
+	ShowUsersStatementExecutor             ShowUsersStatementExecutor
+
+	// Defaults statements to use default database & retention policy
+	// if they are not specified in measurements.
+	StatementNormalizer StatementNormalizer
+
+	// Rewrites statements into other statements, as needed.
+	StatementRewriter *StatementRewriter
```

Comment:
Why is this exposed and a pointer, or even a named type? `StatementRewriter` is defined as an empty struct, and the (pointer) receiver value is never used, other than to call methods. This field should be taken out and non-exported functions should be used instead.